handlers/event_stream_handler_test.go

```diff
@@ -43,16 +43,20 @@ var _ = Describe("EventsHandler", func() {
 		handler = *handlers.NewEventStreamHandler(fakeClient, database, logger, stats)
 	})
 
-	AfterEach(func(done Done) {
-		if server != nil {
-			go func() {
-				server.CloseClientConnections()
-				server.Close()
+	AfterEach(func() {
+		done := make(chan interface{})
+		go func() {
+			if server != nil {
+				go func() {
+					server.CloseClientConnections()
+					server.Close()
+					close(done)
+				}()
+			} else {
 				close(done)
-			}()
-		} else {
-			close(done)
-		}
+			}
+		}()
+		Eventually(done).Should(BeClosed())
```

Comment:
I wonder if this should have a longer timeout than the default?

Reply:
We went with the same timeout that the aftereach used. It seemed to be ok, but if we run into flakes we can bump this for sure.